aws/rust-runtime/aws-config/src/imds/client.rs

```diff
@@ -503,9 +506,8 @@ impl EndpointSource {
                     profile.get(profile_keys::ENDPOINT).map(Cow::Borrowed)
                 };
                 if let Some(uri) = uri_override {
-                    return Ok(
-                        Uri::try_from(uri.as_ref()).map_err(BuildErrorKind::InvalidEndpointUri)?
-                    );
+                    return Ok(Uri::try_from(uri.as_ref())
+                        .map_err(|err| BuildErrorKind::InvalidEndpointUri(err.into()))?);
```

Comment:
These lines are a bit hard for me to parse.

Reply:
Added methods to `BuildError` to construct these, which makes it a lot easier to grok.